index.js

```diff
@@ -50,7 +50,12 @@ class NavigationBar extends Component {
   }
 
   componentWillReceiveProps(props) {
-    customizeStatusBar(this.props.statusBar);
+    if (props.statusBar
+        && ((props.statusBar.hidden !== this.props.statusBar.hidden)
+            || (props.statusBar.style !== this.props.statusBar.style)
```

Comment:
can we move this conditions to a standalone function? Like this:

```js
hasChanged(props) {
  return props.statusBar && ...;
}
```
then our business logic would be pretty clear and straightforward:
```js
componentWillReceiveProps(props) {
  if (hasChanged(props)) {
    customizeStatusBar(props);
  }
}
```